src/main/java/ailurus/Ailurus.java

```diff
@@ -0,0 +1,118 @@
+                    this.ui.sayBye();
+                    isExit = true;
+                    break;
+                case "list":
+                    try {
+                        this.ui.sayList(this.tasks);
+                    } catch (AilurusException e) {
+                        this.ui.showError(e.getMessage());
+                    }
+                    break;
+                case "done":
+                    try {
+                        String str = Parser.parseMessage(fullCommand);
+                        this.ui.sayDone(this.tasks.done(str));
+                    } catch (AilurusException e) {
+                        this.ui.showError(e.getMessage());
+                    }
+                    break;
+                case "todo":
+                    try {
+                        String todoMessage = Parser.parseMessage(fullCommand);
+                        this.ui.sayAdd(this.tasks.addTask(new Todo(todoMessage)), this.tasks.length());
+                    } catch (AilurusException e) {
+                        this.ui.showError(e.getMessage());
+                    }
+                    break;
+                case "deadline":
+                    try {
+                        String deadlineMessage = Parser.parseMessage(fullCommand);
+                        this.ui.sayAdd(this.tasks.addTask(new Deadline(deadlineMessage)), this.tasks.length());
+                    } catch (AilurusException e) {
+                        this.ui.showError(e.getMessage());
+                    }
+                    break;
+                case "event":
+                    try {
+                        String eventMessage = Parser.parseMessage(fullCommand);
+                        this.ui.sayAdd(this.tasks.addTask(new Event(eventMessage)), this.tasks.length());
+                    } catch (AilurusException e) {
+                        this.ui.showError(e.getMessage());
+                    }
+                    break;
+                case "delete":
+                    try {
+                        String str = Parser.parseMessage(fullCommand);
+                        this.ui.sayDelete(this.tasks.deleteTask(str), this.tasks.length());
+                    } catch (AilurusException e) {
+                        this.ui.showError(e.getMessage());
+                    }
+                    break;
+                case "find":
+                    try {
+                        String match = Parser.parseMessage(fullCommand);
+                        this.ui.sayFind(Parser.parseFind(match, this.tasks));
+                    } catch (AilurusException e) {
+                        this.ui.showError(e.getMessage());
+                    }
+                    break;
+                default:
+                    this.ui.sayInvalidCommand();
+                }
+            } catch (AilurusException e) {
+                ui.showError(e.getMessage());
+            }
+        }
+    }
+
```

Comment:
Hi, i think you can put the switch case  to the parser class (parse function) instead. Because that way can divide the functionality of classes, and makes run() shorter.

Reply:
Hi, perhaps you can just use one try-catch block to catch the possible errors since they are all the same kind of exception.

Reply:
Hi, perhaps you can just make use of one try-catch block to deal with the possible errors thrown since they are all the same type of exception.

Reply:
Hi, perhaps you can just make use of one try-catch block to deal with the possible errors thrown since they are all the same type of exception.

Reply:
Thank you for your suggestions! The reason why I placed them in the run function, was because the the ui, storage and tasks are all in the main Ailurus class, and it may not make sense to pass them into a parse function.

Yes thank you! Will remove the try catch blocks.